edition = "2023";

message Tsetlin {
    uint32 n_class   = 1;
    uint32 n_feature = 2;
    uint32 n_clause  = 3;
    uint32 n_state   = 4;

    ModelType model_type = 5;

    repeated Clause clauses = 6;

    enum ModelType {
        INFERENCE = 0;
        TRAINING = 1;
  }
}

message Clause {
    uint32 n_feature = 1;
    uint32 n_state   = 2;

    repeated uint32 data = 3;
}
